Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -296,12 +296,20 @@ void FullscreenManager::cancelFullscreen()
 
     m_pendingExitFullscreen = true;
 
-    m_document.eventLoop().queueTask(TaskSource::MediaElement, [this, topDocument = WTFMove(topDocument), identifier = LOGIDENTIFIER] {
+    m_document.eventLoop().queueTask(TaskSource::MediaElement, [this, weakThis = WeakPtr { *this }, topDocument = WTFMove(topDocument), identifier = LOGIDENTIFIER] {
+        if (!weakThis)
+            return;
+
         if (!topDocument->page()) {
             INFO_LOG(identifier, "Top document has no page.");
             return;
         }
 
+        if (!page()) {
+            ERROR_LOG(LOGIDENTIFIER, "Document no longer in page; bailing");
+            return;
+        }
+
         // This triggers finishExitFullscreen with ExitMode::Resize, which fully exits the document.
         if (auto* fullscreenElement = topDocument->fullscreenManager().fullscreenElement())
             page()->chrome().client().exitFullScreenForElement(fullscreenElement);
```

Comment:
Actually looking more at this, can we be consistent about the page we're using and only use a single check?

We shouldn't need to check for both topDocument->page() and page() at the same time.

I suspect it would be reasonable to call `topDocument->page()->chrome().client().exitFullScreenForElement(fullscreenElement);`